13-cqrs/1-basic/2-after/ShipmentCommand.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * This class is ONLY responsible to mutates the delivery state
+ *
+ * @author Nadeesha Dilruwan <nadeesha@adsdaq.eu>
+ */
+final class ShipmentCommand
+{
+    /**
+     * @var DateTime
+     */
+    private $delivery;
+
+    /**
+     * @var string
+     */
+    private $proofOfDelivery;
+
+    /**
+     * ShipmentCommand constructor.
+     * @param string|UUID $shipmentID
+     */
+    public function __construct($shipmentID)
+    {
+        // Load current state
+    }
+
+    /**
+     * @param DateTime $delivery
+     * @param string $proofOfDelivery
+     */
+    public function Deliver(DateTime $delivery, $proofOfDelivery)
```

Comment:
why is this method in the command?

Reply:
Is not the command simple objects?

Reply:
Or may be this method should be a part of the command handler?

Reply:
yes yes, in the actual implementation this is in the handler.. This basic one I use to explain the separation. got it ?

Reply:
OKay, still confusing, but wait for the explanation that you may provide on Monday